mne/preprocessing/_regress.py

```diff
@@ -53,35 +68,257 @@ def regress_artifact(inst, picks=None, picks_artifact='eog', betas=None,
     ----------
     .. footbibliography::
     """  # noqa: E501
-    _check_preload(inst, 'regress')
-    _validate_type(inst, (BaseEpochs, BaseRaw), 'inst', 'Epochs or Raw')
-    picks = _picks_to_idx(inst.info, picks, none='data')
-    picks_artifact = _picks_to_idx(inst.info, picks_artifact)
-    if np.in1d(picks_artifact, picks).any():
-        raise ValueError('picks_artifact cannot be contained in picks')
-    inst = inst.copy() if copy else inst
-    artifact_data = inst._data[..., picks_artifact, :]
-    ref_data = artifact_data - np.mean(artifact_data, -1, keepdims=True)
-    if ref_data.ndim == 3:
-        ref_data = ref_data.transpose(1, 0, 2).reshape(len(picks_artifact), -1)
-    cov = np.dot(ref_data, ref_data.T)
-    # process each one separately to reduce memory load
-    betas_shape = (len(picks), len(picks_artifact))
     if betas is None:
-        betas = np.empty(betas_shape)
-        estimate = True
+        model = EOGRegression(picks=picks, exclude=exclude,
+                              picks_artifact=picks_artifact, proj=proj)
+        model.fit(inst)
     else:
-        estimate = False
-        betas = np.asarray(betas, dtype=float)
-        _check_option('betas.shape', betas.shape, (betas_shape,))
-    for pi, pick in enumerate(picks):
-        this_data = inst._data[..., pick, :]  # view
-        orig_shape = this_data.shape
-        if estimate:
-            # subtract mean over time from every trial/channel
+        # Create an EOGRegression object and load the given betas into it.
+        picks = _picks_to_idx(inst.info, picks, exclude=exclude, none='data')
+        picks_artifact = _picks_to_idx(inst.info, picks_artifact)
+        want_betas_shape = (len(picks), len(picks_artifact))
+        _check_option('betas.shape', betas.shape, (want_betas_shape,))
+        model = EOGRegression(picks, picks_artifact, proj=proj)
+        all_picks = np.unique(np.hstack((picks, picks_artifact)))
+        model.info_ = pick_info(inst.info, all_picks)
+        model.coef_ = betas
+    return model.apply(inst, copy=copy), model.coef_
+
+
+@fill_doc
+class EOGRegression():
+    """Remove EOG artifact signals from other channels by regression.
+
+    Employs linear regression to remove signals captured by some channels,
+    typically EOG, but it also works with ECG from other channels, as described
```

Comment:
This contradicts what the tutorial, where you say that it doesn't really work for removing ECG. I think we should be consistent.

Reply:
Let's remove this sentence from the docs.